Automattic/wp-calypso
# Site Editor Domain Upsell: Add callout using ETK V2

Fixes https://github.com/Automattic/wp-calypso/issues/76321
Fixes https://github.com/Automattic/wp-calypso/issues/73445

This is a new version of [this PR](https://github.com/Automattic/wp-calypso/pull/76855), because it was reverted (p1684250921623969-slack-C02FMH4G8)

## Proposed Changes
Add Domain Upsell callout to Gutenberg using [editing toolkit](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/README.md)

#### Conditions to show the callout
- User has a verified email
- Site has no custom domain
- Site is not a P2
- Not show if has `siteIntent` (like start-writing, link-in-bio, etc)

#### Track events
- View: `calypso_block_editor_domain_upsell_callout_view`
- Click: `calypso_block_editor_domain_upsell_callout_click`
- Dismiss: `calypso_block_editor_domain_upsell_callout_dismiss`

## Testing Instructions

- You need to sandbox your testing site domain
- Go to `/apps/editing-toolkit`
- Run `yarn dev --sync`
- On Calypso, go to `/posts` (and `/pages`) and edit or create new.
- Or, change the view to classic and use `/wp-admin`
- You should see the callout if conditions matches.
- Check CTA
- Check dismiss (you can re-enable it by running `wp.data.dispatch( 'core/preferences' ).set( 'core/edit-post', 'domain_upsell_callout_dismiss', false );` in the console and reloading the page.
- Check track events

To test on Atomic Sites, follow the instructions here: PCYsg-ly5-p2 (**ETK and Atomic Sites**)

## Screenshots
![image](https://github.com/Automattic/wp-calypso/assets/402286/9aabc708-d68d-4349-b118-c1fd0934591f)
![image](https://github.com/Automattic/wp-calypso/assets/402286/e8722d83-ba3a-4eb1-b6bc-40b85c0a51e6)
![image](https://github.com/Automattic/wp-calypso/assets/402286/1808d8cb-9268-45ef-b10e-904e3df2f7a8)
![image](https://github.com/Automattic/wp-calypso/assets/402286/e0f3a5fc-6f64-4007-946f-dc784a1e1554)
![image](https://github.com/Automattic/wp-calypso/assets/402286/fe703af4-46d9-4e90-bf6c-00153f79b9bd)


## Pre-merge Checklist
